### ISSUE TITLE
Wrong text insert in CodeArea

### ISSUE THREAD

**Issue by ErnstMosinski** [TomasMikula/RichTextFX#475]
I just tested the CodeArea and downloaded the richtextfx-0.7-M4 JARs.
If you enter text in the CodeArea the cursor is not moved and all characters are enter at the current position.
This looks like you type in text in reverse.
This can easily be reproduce by running the JavaKeywords example from the richtextfx-demos-fat-0.7-M4.jar file.
Is it possible to workaround this by setting some properties, or is it just a bug who needs to be fixed?

**Comment by kuc** [comment]
Duplicate of #472, already fixed.

**Comment by JordanMartinez** [comment]
It is possible to fix this regression in the `0.7-M4` release by overriding the `replace` method like so:
````java
public void replace(int from, int to, StyledDocument<PS, SEG, S> document) {
    super.replace(from, to, document);
    int newCaretPos = getCaretPosition() + document.length();
    selectRange(newCaretPos, newCaretPos);
}
````

**Comment by ErnstMosinski** [comment]
Thanks for this information!!
